src/Analysis/pipeline/parameter_types.py

```diff
@@ -0,0 +1,323 @@
+import abc
+
+class BaseData(metaclass=abc.ABCMeta):
+    """
+    Abbstract class od port types, that define user for
+    validation and translation to string
+    """
+    def __init__(self):
+        self.name = ""
+        """Display name of port"""
+        self.description = ""
+        """Display description of port"""
+    
+    @abc.abstractmethod 
+    def to_string(self, value):
```

Comment:
Jak jsem naznačil v popisu, jde o převod parametru pro potřebu vložení do yaml souboru ve flow stylu. Tento převod je jednosměrný (druhý směr není asi potřeba) a není to serializace, abych to dlišil jsem ti nenazval ani _serialize ani dump